src/genie/libs/parser/ironware/show_mpls.py

```diff
@@ -0,0 +1,691 @@
+
+        # End-Point state  : Up
+        p2 = re.compile(r'(^End-Point state\s+:\s+(?P<state>Up|Down))')
+
+        # MCT state        : None
+        p3 = re.compile(r'(^MCT state\s+:\s+(?P<mctstate>\S+$))')
+
+        # IFL-ID           : --
+        # IFL-ID           : n/a
+        # IFL-ID           : 1234
+        p4 = re.compile(r'(^IFL-ID\s+:\s+(?P<iflid>n\/a$|\d+$|--$))')
+
+        # Local VC type    : tag
+        # Local VC type : raw-pass-through
+        p5 = re.compile(r'(^Local VC type\s+:\s+'
+                        r'(?P<vctype>tag$|raw-pass-through$|--$|raw-mode$))')
+
+        # Local VC MTU     : 9190
+        p6 = re.compile(r'(^Local VC MTU\s+:\s+(?P<mtu>\d+$|--$))')
+
+        # COS              : --
+        p7 = re.compile(r'(^COS\s+:\s+(?P<cos>\S+$))')
+
+        # Extended Counters: Enabled
+        p8 = re.compile(r'(^Extended Counters:\s+'
+                        r'(?P<extcounters>[e|E]nabled|[d|D]isabled)$)')
+
+        # Counter          : disabled
+        p9 = re.compile(r'(^Counter\s+:\s+'
+                        r'(?P<counter>[e|E]nabled|[d|D]isabled)$)')
+
+        # Vll-Peer         : 192.168.1.1
+        p10 = re.compile(r'(^Vll-Peer\s+:\s+'
+                         r'(?P<ip>\d{1,3}.\d{1,3}.\d{1,3}.\d{1,3}$))')
+
+        # State          : UP
+        # State          : DOWN - PW is Down (Reason:Wait for peer label)
+        # For all states see page 550 in reference material
+        p11 = re.compile(r'(^State\s+:\s+(?P<state>UP|DOWN)(\s-\s'
+                         r'(?P<reason>\w.+$)|$))')
+
+        # Remote VC type : tag               Remote VC MTU  : 9190
+        p12 = re.compile(r'(^Remote VC type\s+:\s+'
+                         r'(?P<vctype>tag|raw-pass-through|--)\s+'
+                         r'Remote VC MTU\s+:\s+(?P<mtu>\d+$|--$))')
+
+        # Local label    : 852217            Remote label   : 852417
+        # Local label    : --                Remote label   : --
+        p13 = re.compile(r'(^Local label\s+:\s(?P<local>\d+|--)\s+'
+                         r'Remote label\s+:\s+(?P<remote>\d+$|--$))')
+
+        # Local group-id : 0                 Remote group-id: 0
+        p14 = re.compile(r'(^Local group-id\s+:\s+(?P<localgid>\d+|--)\s+'
+                         r'Remote group-id:\s+(?P<remotegid>\d+$|--$))')
+
+        # Tunnel LSP     : mlx8.1_to_ces.2 (tnl15)
+        p15 = re.compile(r'(^Tunnel LSP\s+:\s+'
+                         r'(?P<lspname>[^+\s]+)\s+\((?P<tunnel>\w+)\)$)')
+
+        # LSPs assigned  : No LSPs assigned
+        p16 = re.compile(r'(^LSPs\s+assigned\s+:\s+(?P<lsps>\w+.*$))')
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            m = p0.match(line)
+            if m:
+                vll_dict = result_dict.setdefault('vll', {}).setdefault(vll, {
+                    'vcid': int(m.groupdict()['vcid']),
+                    'vll_index': int(m.groupdict()['vllindex'])
+                })
+                continue
+
```

Comment:
@jamesditrapani  You havent addressed this comment.
It is necessary to have these comments since it is large parser. When re-reading it would help us see the line example right there. 
So please add a comment on top of all line similar to below
```
m = p2.match(line)

Reply:
@KamyarZiabari Sorry, looks like I missed this one originally!